### PR TITLE
Remote values and templating rendering fix

### PR DESCRIPTION
**Pull request by kuzaxak**
I solved an [issue][1] with templating a single chart with dependency. And added ability to use remote values in release that was requested few times: [one][2], [two][3]. 

Please check commit messages for additional details.

[1]: https://github.com/roboll/helmfile/issues/2048
[2]: https://github.com/roboll/helmfile/issues/1838
[3]: https://github.com/roboll/helmfile/issues/1499

### CODE REVIEW

**Comment by kuzaxak** [comment]
@mumoshu Hey, would you be able to review it this or the next week? We would like to understand what approach to choose in developing a new pipeline.

**Comment by ffoysal** [comment]
it is really helpful feature. we are planning to use the values files same way

**Comment by indrekj** [comment]
This would be useful

File: pkg/app/app.go [hunk 1]
```diff
@@ -1830,6 +1830,24 @@ func (a *App) template(r *Run, c TemplateConfigProvider) (bool, []error) {
...
+	var traversedReleasesWithNeeds []state.ReleaseSpec
+	for _, rs := range batches {
+		for _, r := range rs {
+			traversedReleasesWithNeeds = append(traversedReleasesWithNeeds, r.ReleaseSpec)
+		}
+	}
+
+	// Traverse DAG of all the releases so that we don't suffer from false-positive missing dependencies
+	// withDAG require dependencies to be present, in case of strict selector we should include them anyway
+	st.Releases = traversedReleasesWithNeeds
```
Comment (mumoshu) [inline pkg/app/app.go:1849]: @kuzaxak Hey! Thanks for the PR. The other half of your changes looks good. But I can't see why this one is needed.
Saw your `template_a_single_release_with_dependency` test case- shouldn't you just specify `--skip-needs` if you don't want to include the dep?

Comment (kuzaxak) [reply pkg/app/app.go:1849]: Hey, we discussed that in your PR [here][1] but this PR closed

[1]: https://github.com/helmfile/helmfile/pull/78

Comment (kuzaxak) [reply pkg/app/app.go:1849]: @mumoshu I expect yes, but in any case I will test it right after merge. Based on your tests cases it should work as you described.

I will rebase and delete commit with graph changes.

Comment (mumoshu) [reply pkg/app/app.go:1849]: @kuzaxak #78 has been merged. Would you mind giving it a shot and remove the graph changes if it looks good?

Comment (kuzaxak) [reply pkg/app/app.go:1849]: Cool, yes, I will rebase my branch and remove graph related changes.

Comment (kuzaxak) [reply pkg/app/app.go:1849]: I'm not even sure is it needed anymore due to merged [PR][1]. Will test and close if not needed. 

[1]: https://github.com/helmfile/helmfile/pull/47

**Comment by kuzaxak** [comment]
Works as expected in 0.145

**Pull request closed without merging**